test/layout/key_test.exs

```diff
@@ -0,0 +1,28 @@
+defmodule LayoutKeyTest do
+  use ExUnit.Case
+
+  alias Layout.Key
+
+  test "new/3 creates a %Key{} struct with default :width, :height, and :led" do
+    assert Key.new(:a, 0, 0) == %Key{id: :a, x: 0, y: 0, width: 1, height: 1, led: nil}
+    assert Key.new(:b, 1, 2) == %Key{id: :b, x: 1, y: 2, width: 1, height: 1, led: nil}
+  end
+
+  test "new/4 allows setting :width, :height, and :led" do
```

Comment:
If you break this out into 3 separate tests (width, height, led), it'll decouple the tests for this functionality.

Or if width + height _must_ be set together, then I'd omit LED from this test and add other tests for situations like "height must be set if width is set".

Reply:
I actually do break it into three separate tests below, and I should probably just remove this one...